otherlibs/unix/read_unix.c

```diff
@@ -39,16 +39,14 @@ CAMLprim value caml_unix_read(value fd, value buf, value ofs, value len)
 
 CAMLprim value caml_unix_read_bigarray(value fd, value vbuf, value vofs, value vlen)
 {
-  CAMLparam0();
+  CAMLparam1(vbuf);
   intnat ofs, len;
-  void *buf;
   int ret;
 
-  buf = Caml_ba_data_val(vbuf);
   ofs = Long_val(vofs);
   len = Long_val(vlen);
   caml_enter_blocking_section();
-  ret = read(Int_val(fd), buf + ofs, len);
+  ret = read(Int_val(fd), Caml_ba_data_val(vbuf) + ofs, len);
```

Comment:
Color me confused, isn't this change *introducing* a GC safety bug? Once you enter a blocking section, another thread may perform a GC and move `vbuf` under your feet. See for example

https://github.com/ocaml/ocaml/blob/5ff9ea2780bcae5c1bbc675c9e2f341e080cba42/otherlibs/unix/lseek_unix.c#L57-L63

Reply:
> Color me confused, isn't this change _introducing_ a GC safety bug? Once you enter a blocking section, another thread may perform a GC and move `vbuf` under your feet.

I was convinced that registering `vbuf` as a root was enough to guard against that, but now am not so sure anymore.